python/lsst/jointcal/jointcal.py

```diff
@@ -277,32 +363,46 @@ def run(self, dataRefs, profile_jointcal=False):
                 print("still some ouliers but chi2 increases - retry")
             else:
                 break
-                print("unxepected return code from Minimize")
+                print("unxepected return code from minimize")
 
-        # Fill reference and measurement n-tuples for each tract
-        tupleName = "res_" + str(dataRefs[0].dataId["tract"]) + ".list"
-        fit.MakeResTuple(tupleName)
+        return fit, model, sky2TP
 
-        # Build an updated wcs for each calexp
-        imList = associations.TheCcdImageList()
+    def _write_results(self, associations, astrom_model, photom_model, dataRefs):
+        """
+        Write the fitted results (photometric and astrometric) to a new 'wcs' dataRef.
+
+        Parameters
+        ----------
+        associations : lsst.jointcal.Associations
+            The star/reference star associations to fit.
+        astrom_model : lsst.jointcal.AstromModel
+            The astrometric model that was fit.
+        photom_model : lsst.jointcal.PhotomModel
+            The photometric model that was fit.
+        dataRefs : list of lsst.daf.persistence.ButlerDataRef
+            List of data references to the exposures that were fit.
+        """
 
-        for im in imList:
-            tanSip = spm.ProduceSipWcs(im)
-            frame = im.ImageFrame()
+        ccdImageList = associations.getCcdImageList()
+        for ccdImage in ccdImageList:
+            tanSip = astrom_model.ProduceSipWcs(ccdImage)
+            frame = ccdImage.ImageFrame()
             tanWcs = afwImage.TanWcs.cast(jointcalLib.GtransfoToTanWcs(tanSip, frame, False))
 
-            name = im.Name()
+            # TODO: there must be a better way to identify this ccdImage?
+            name = ccdImage.Name()
             visit, ccd = name.split('_')
             for dataRef in dataRefs:
-                ccdname = dataRef.get("calexp").getDetector().getId()
+                calexp = dataRef.get("calexp")
+                ccdname = calexp.getDetector().getId()
                 if dataRef.dataId["visit"] == int(visit) and ccdname == int(ccd):
                     print("Updating WCS for visit: %d, ccd%d" % (int(visit), int(ccd)))
-                    exp = afwImage.ExposureI(0, 0)
-                    exp.setWcs(tanWcs)
+                    exp = afwImage.ExposureI(0, 0, tanWcs)
+                    exp.setCalib(calexp.getCalib())  # start with the original calib
+                    fluxMag0 = calexp.getCalib().getFluxMag0()  # returns (fluxMag0, fluxMag0Err)
```

Comment:
Good point. I've reworked this. The error I'm not certain about, since jointcal doesn't fit the error term, but I can just pass on the error from the original.